### PR TITLE
Document sources and prep macros

### LINKED ISSUE #45: Document other important terms in the README
- Explain sources
  - local/remote
  - patches are sources too
  - indices, how they work, implicit indexing
  - `%sourcelist`/`%patchlist`

- `%prep` section and macros recognized there

### PR DESCRIPTION
**Pull request by nforro**
Fixes #45.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -24,6 +24,20 @@ Tag represents a single item of metadata of a package. It has a well-defined nam
 
 For the purposes of this library, a tag can have associated comments. These are consecutive comment lines directly above the tag definition in a spec file.
 
+### Source
+
+Source is a source file or a downstream patch defined by a `Source`/`Patch` tag or by an entry in `%sourcelist`/`%patchlist` section.
+
+Source can be local, specified by a filename, or remote, specified by a URL. Local sources should be located in and remote sources should be downloaded to a directory referred to as `sourcedir`.
```
Comment (FrNecas) [inline README.md:31]: The wording of the second sentence is a bit weird, at first when reading through it, my first thought was "in what?" and I realized that the rest is at the end. Maybe splitting it into 2 sentences would make it better? `Local sources should be located in a directory referred to as sourcedir. Remote sources should be downloaded to this directory.`

File: README.md [hunk 2]
```diff
@@ -24,6 +24,20 @@ Tag represents a single item of metadata of a package. It has a well-defined nam
...
+
+Sources defined by tags can be explicitly numbered, e.g. `Source0` or `Patch999`, otherwise implicit numbering takes place and source numbers are auto-assigned in a sequential manner.
+
+### Prep macros
+
+Prep macros are macros that often appear in (and only in, they don't make sense anywhere else) `%prep` section.
+
+4 such macros are recognized by this library, `%setup`, `%patch`, `%autosetup` and `%autopatch`. A typical spec file uses either `%autosetup` or a combination of `%setup` and `%patch` or `%autopatch`.
```
Comment (TomasTomecek) [inline README.md:39]:
```suggestion
4 such macros are recognized by this library, [`%setup`](https://rpm-packaging-guide.github.io/#setup), `%patch`, `%autosetup` and `%autopatch`. A typical spec file uses either `%autosetup` or a combination of `%setup` and `%patch` or `%autopatch`.
```

**Review by TomasTomecek — APPROVED** [review]
Added a reference for %setup in the RPM packaging guide.

Thanks for writing this up!

File: README.md (hunk 1 above)
Comment (nforro) [reply README.md:31]: Ok, I'll try to do something about it.

File: README.md (hunk 2 above)
Comment (nforro) [reply README.md:39]: Ok, maybe we should add http://ftp.rpm.org/max-rpm/s1-rpm-inside-macros.html which documents also `%patch`, and https://rpm-software-management.github.io/rpm/manual/autosetup.html for `%autosetup` and `%autopatch`.

**Pull request merged**